File: XMLCheck/Handler/CheckRulesXML10/Xml10RuleBase.cs
using XMLCheck.Models;
using XMLCheck.Services;

namespace XMLCheck.Handler.CheckRulesXML10
{
    public abstract class Xml10RuleBase:IXml10Rule
    {
        public abstract string Key { get; }

        public abstract ErrorDetails? Check(Xml10Model model,HoSoContext context);

        protected ErrorDetails Error(string message)
        {
            return new ErrorDetails
            {
                key = Key,
                message = message
            };
        }
    }
}

File: XMLValidator_v2/src/parsers/rules/xml10/xml10RuleBase.ts
import { Xml10Model } from '../../../interfaces/xml-models.interface';
import { HoSoContext } from '../../../services/hoso-context';
import { ErrorDetails } from '../../../dto/xml-check.dto';
import { BaseRule } from '../base.rule';

export interface IXml10Rule {
  readonly key: string;
  readonly isEnabled: boolean;
  check(model: Xml10Model, context: HoSoContext): ErrorDetails | null;
}

export abstract class Xml10RuleBase extends BaseRule<Xml10Model> implements IXml10Rule {}
